pkg/acs/calacs/acsccd/blev_funcs_postsm4.c

```diff
@@ -36,152 +40,243 @@ static int make_amp_array(const int arr_rows, const int arr_cols, SingleGroup *
 static int unmake_amp_array(const int arr_rows, const int arr_cols, SingleGroup * im,
                             int amp, double * array);
 
+/* Remove the signal dependent bias shift from post-SM4 full frame WFC data
+ * (based on ISR http://www.stsci.edu/hst/acs/documents/isrs/isr1202.pdf),
+ * as well as from supported subarray data (~post-mid-2016) as identified by
+ * specific aperture names 
+ * (based on ISR http://www.stsci.edu/hst/acs/documents/isrs/isr1703.pdf).
+ *
+ * chip2 consists of amps C & D, chip1 consists of amps A & B. 
+ *
+ * Because full frame data uses both chips, but supported subarray data
+ * is only from a single amp (therefore, a portion of a single chip), this
+ * routine has a variable number of input SingleGroups.
+ *
+ * *** This routine should be modified according to issues discussed in PR #312 and noted in IT #334. ***
+ *
+ * 16-May-2018 M.D. De La Pena: Generalized bias_shift_corr() to handle subarray data.
+ */
+int bias_shift_corr(ACSInfo *acs, int nGroups, ...) {
+
+  /* This function has built in assumptions regarding the data being passed */
+  assert (acs != NULL);  /* Make sure the ACSInfo structure is defined */
+  assert ((nGroups == 1) || (nGroups == 2)); /* Must be at least one and no more than two SingleGroups */
+  assert (((nGroups == 1) && (acs->subarray == YES)) || ((nGroups == 2) && (acs->subarray == NO)));  /* Subarray has one and full frame has two SingleGroups */
+
+  /* Code to handle variable number of input parameters.  The number of SingleGroup is 2 when 
+     processing a full frame (2 chips - all 4 amps) and only 1 when processing a supported subarray 
+     (1 chip from a single amp).
+  */
+  SingleGroup *sg[nGroups];
+  va_list arguments;
+
+  /* Initializing arguments to store all the values being passed to this routine via "...".
+   * The nGroups value and the number of SingleGroups actually passed must match. 
+   */
+  va_start (arguments, nGroups);
+  {
+       /* sg[0] is chip2 and sg[1] is chip 1 for a full frame image */
+       {unsigned int i;
+       for (i = 0; i < nGroups; i++) {
+           sg[i] = NULL;
+           sg[i] = va_arg (arguments, SingleGroup *);
+           if (!sg[i]) {
```

Comment:
This is not what I had in mind. I meant that if ``va_arg (arguments, SingleGroup *) == NULL`` it just wouldn't be added, i.e. 
```c
SingleGroup * tmpGroup = va_arg (arguments, SingleGroup *);
if (tmpGroup)
    sg[i] = tmpGroup;
else
    continue;
```
You could argue that in such a case ``nGroups`` is actually incorrect, it is unclear of the use of the function params now that they are variadic. I.e. are they explicit slots, e.g. ``SingleGroup * amp2, SingleGroup * amp1`` or is the order irrelevant and in the case that there is only one, it must always be the first, e.g. not passed as ``NULL, amp1``.

Reply:
I think the rest of the code assumes a specific ordering.